listener.c

```diff
@@ -8,7 +8,7 @@
 
 /* This needs to be recursive safe because we may log to clients
  * while we are dispatching subscriptions to clients */
-pthread_mutex_t w_client_lock;
+pthread_mutex_t w_client_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
technically this one needs a recursive initializer, but its okay because we guarantee initialization below (I'll add a line note in a moment).
Please remove this line from this PR as it doesn't contribute to solving the root of the problems :-)